### DISCUSSION TITLE
Would an WebGL package work in mobile?

### DISCUSSION THREAD

**Discussion by keremnymn** [tauri-apps/tauri#7966] [Q&A] [answered]
Hi,

I am planning to build a desktop app using Tauri and one of the libraries that I will use is dependent on WebGL (Maplibre GL JS). And also I know that in Tauri 2.0 mobile development is targeted as well.

Since I am mostly from a web development background, I am curious if the app I am developing would work on mobile. Because I know that native mobile apps use different GPU APIs. If Tauri 2.0 mobile support will behave as WebView, I believe it would work, but I am not sure.

I positive that I am missing some points here, any ideas and extra information would be much appreciated.

**Comment by FabianLars** [comment, marked as answer]
Generally webgl works on mobile (both android and ios), yes. It shouldn't require any special care on your site and work out of the box.

**Comment by xixioba** [comment]
Hi @FabianLars 
I have a similar but special need. On the desktop side, I added an axum static file server in tauri to support the front-end using **iframe** to display unity webgl, so that I can package resource files and publish them to a non-network environment.
Is this also possible for ios or android? If not, is there any way I can provide similar functions without providing network services?
Thanks

**Reply by FabianLars** [reply]
Hi.

I don't think you can spawn a server on mobile but i'm not 100% sure. Did you ever try the `asset:` protocol (via <https://tauri.app/v1/api/js/tauri/#convertfilesrc>) for your iframe content? I thought that'd work but also never tried it myself.
Can't think of anything else right now.
